### PR TITLE
Wcslib 5.10

### PR DESCRIPTION
**Pull request by nden**
This updates wcslib to v 5.10.

### CODE REVIEW

File: cextern/wcslib/CHANGES [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+
+* C library
+
+  - In wcshdo(), allow output of floating point keyvalues to 15
+    significant digits (was 14), mainly for astropy.
+
+  - In wcspih(), allow unrecognised WAT projection types (tan, etc.) to
+    pass without returning an error.  Reported by Michael Droettboom.
+    Also fixed a bug triggered by SIP keywords in unconventional order,
+    reported by Colin Slater, LSST.
```
Comment (saimn) [inline cextern/wcslib/CHANGES]: Great, this should fix #4089

Comment (nden) [inline cextern/wcslib/CHANGES]: Yes, I've confirmed this fixes #4089. Any chance it can go in 1.1? Or is it too late?
 @embray @astrofrog

**Comment by embray** [comment]
I don't think it's too late, but I leave that to @eteq and @astrofrog to decide.  I think we might as well though--I think we can make an exception given that we were waiting on upstream here...

There's also some line in the changelog (that keeps having to be updated!) specifying which version of WCSLIB Astropy v1.1 is upgrading to.  So that should be updated again :)

**Comment by astrofrog** [comment]
I think we should include it in 1.1 since we included a patch in https://github.com/astropy/astropy/pull/4045 that would be cleaner to include with the real 5.10.

**Comment by astrofrog** [comment]
@embray - fixed the changelog in https://github.com/astropy/astropy/commit/f23a545bb08b20cf8fa70a9adb41ba9e79e7d79e

**Pull request merged**